collie/metrics.py

```diff
@@ -277,7 +277,10 @@ def auc(targets: csr_matrix,
 
 
 def evaluate_in_batches(
-    metric_list: Iterable[Callable[..., Any]],
+    metric_list: Iterable[Callable[
+        [csr_matrix, Union[np.array, torch.tensor], Union[np.array, torch.tensor], Optional[int]],
```

Comment:
[Extra credit] This is NOT your problem at all, but looking at this, I realize that metrics (like `auc` above) have a typo. Would you be willing to replace something like: 

```python
user_ids: (np.array, torch.tensor),
preds: (np.array, torch.tensor),
```

in that function definition with: 

```python
user_ids: Union[np.array, torch.tensor],
preds: Union[np.array, torch.tensor],
```

?

Reply:
Happy to!